File: python/_modulos/fila.py
class Fila:
    def __init__(self):
        self.max = 100
        self.fila = [None]*self.max
        self.fim = 0
        self.inicio = 0

    def inserir(self, valor):
        if self.fim == self.max:
            self.max *= 2
            temp = [None]*self.max

            for i in range(self.inicio, int(self.max / 2)):
                temp[i - self.inicio] = self.fila[i]

            self.fila = temp
            self.inicio = 0
            self.fila[self.fim] = valor
        else:
            self.fila[self.fim] = valor

        self.fim += 1

    def remover(self):
        if self.fim == self.inicio:
            print("Fila vazia")
        else:
            elemento = self.fila[self.inicio]
            self.inicio += 1
            return elemento
        
    def imprimir(self):
        for i in range(self.inicio, self.fim):
            print(self.fila[i], end=" ")

    def tamanho(self):
        return self.fim - self.inicio

File: javascript/20_fila/fila.js
const fila = function() {
	var max = 100;
	var fila = new Array(max);
	var inicio = 0;
	var fim = 0;
	
	function inserir(valor) {
		if (fim == max) {
			max *= 2;
			var temp = new Array(max);
			
			for (var i = inicio; i < max / 2; i++) {
				temp[i - inicio] = fila[i];
			}
			
			fila = temp;
			inicio = 0
			fila[fim] = valor;
		} else {
			fila[fim] = valor;
		}
		
		fim++;
	}
	
	function remover() {
		if (fim == inicio) {
			document.write("Fila vazia<br>");
		} else {
			var elemento = fila[inicio];
			inicio++;
			return elemento;
		}
	}
	
	function imprimir() {
		for (var i = inicio; i < fim; i++) {
			document.write(fila[i] + " ");
		}
	}
	
	function tamanho() {
		return fim - inicio;
	}
	
	return {
		inserir: inserir,
		remover: remover,
		imprimir: imprimir,
		tamanho: tamanho
	}
}();